py/server/tests/test_table_data_service.py

```diff
@@ -357,9 +360,17 @@ def test_partition_size_sub_failure(self):
         data_service = TableDataService(backend)
         backend.sub_partition_size_fail_test = True
         table = data_service.make_table(TableKeyImpl("test"), refreshing=True)
+        table.coalesce()
+        # wait for the size subscription failure callback to be called for 5 seconds
+        for _ in range(50):
+            time.sleep(0.1)
+            if backend.is_size_sub_failure_cb_called:
+                break
+        else:
+            self.fail("size subscription failure callback was not called in 5s")
+
         with self.assertRaises(Exception) as cm:
-            # failure_cb will be called in the background thread after 2 PUG cycles, 3 seconds timeout should be enough
-            self.wait_ticking_table_update(table, 600, 3)
+            self.wait_ticking_table_update(table, 600, 1)
```

Comment:
I suspect you need 2 seconds (2 cycles) to be safe if you're using a real `PeriodicUpdateGraph` with a 1 second interval. The failure is buffered after the roots are processed on one cycle, it won't be delivered until the next.